packages/react-components/src/components/RichTextEditor/RichTextSendBox.tsx

```diff
@@ -246,19 +294,47 @@ export const RichTextSendBox = (props: RichTextSendBoxProps): JSX.Element => {
   }, [contentValue]);
 
   /* @conditional-compile-remove(file-sharing-acs) */
-  const toAttachmentMetadata = useCallback((attachmentsWithProgress: AttachmentMetadataInProgress[] | undefined) => {
-    return attachmentsWithProgress
-      ?.filter((attachment) => {
-        return !('error' in attachment) && !attachment.error?.message;
-      })
-      .map((attachment) => {
-        return {
-          id: attachment.id,
-          name: attachment.name,
-          url: attachment.url ?? ''
-        };
+  /* @conditional-compile-remove(rich-text-editor-image-upload) */
+  const toAttachmentMetadata = useCallback(
+    (
+      attachmentsWithProgress: AttachmentMetadataInProgress[] | undefined,
+      /* @conditional-compile-remove(rich-text-editor-image-upload) */ attachmentType?: 'image'
+    ) => {
+      return attachmentsWithProgress
+        ?.filter((attachment) => {
+          return !('error' in attachment) && !attachment.error?.message;
+        })
+        .map((attachment) => {
+          let url = attachment.url || '';
+          /* @conditional-compile-remove(rich-text-editor-image-upload) */
+          if (attachmentType === 'image') {
+            url = '';
+          }
+          return {
+            id: attachment.id,
+            name: attachment.name,
+            url,
+            /* @conditional-compile-remove(rich-text-editor-image-upload) */
+            attachmentType: attachmentType
+          };
+        });
+    },
+    []
+  );
+
+  /* @conditional-compile-remove(rich-text-editor-image-upload) */
+  const addUploadedImagesToMessage = useCallback(
```

Comment:
It might be a good idea to addd comment about why we need to do this (for better support later)